File: 800/1462A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t; // number of test cases
    
    while (t--) 
    {
        int n;
        cin >> n; // size of the array
        vector<int> a(n);
        
        for (int i = 0; i < n; i++) 
        {
            cin >> a[i];
        }
        
        int left = 0;
        int right = n - 1;
        
        // Print elements alternately from left and right
        while (left < right) 
        {
            cout << a[left] << " " << a[right] << " ";
            left++;
            right--;
        }
        
        // If odd length, print the middle element
        if (left == right) 
        {
            cout << a[left] << " ";
        }
        
        cout << "\n";
    }
    
    return 0;
}

File: 800/1462A/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        
        int t = sc.nextInt();
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int a[] = new int[n];
            
            for(int i=0;i<n;i++)
                a[i] = sc.nextInt();
            
            int left = 0;
            int right = n-1;
            
            while(left < right)
            {
                System.out.print(a[left]+" "+a[right]+" ");
                left++;
                right--;
            }
            
            if(left == right)
                System.out.print(a[left]+" ");
            
            System.out.println();
        }
    }
}